### DISCUSSION TITLE
How can I hightlight specific conduits?

### DISCUSSION THREAD

**Discussion by CedarSnowy** [pyswmm/swmmio#198] [Q&A] [answered]
I want to highlight  a couple of specific conduits in the form of animations or image,how can I do this?

**Comment by bemcdonnell** [comment]
@CedarTH are you using the swmmio mapping features? Assuming so, I’m going to migrate this over to swmmio

**Reply by CedarSnowy** [reply]
Yes,I am using swmmio.create_map from demo "Latte_art_double_shot".It seemed that I can hightlight specific conduits by editing the output HTML file,but I haven't success yet

**Reply by aerispaha** [reply, marked as answer]
Hey @CedarTH, currently the `swmmio.create_map` function is not super flexible, unless you're comfortable playing around with Javascript and the [Mapbox GL JS](https://www.mapbox.com/mapbox-gljs) library. To customize the map, you can pass a path to your own basemap file, in which you can customize the Mapbox visualization logic. If you'd like to do this, I recommend starting by modifying a copy of the default basemap file used by `swmmio.create_map`, which you can download from [here](https://github.com/pyswmm/swmmio/blob/master/swmmio/reporting/basemaps/mapbox_base.html). 

With a custom basemap, you can run the `create_map` function like this: 

```python 
mymodel = swmmio.Model('path/to/model.inp', crs='MYCRSCODE')

swmmio.create_map(mymodel, basemap='path/to/my/basemap.html')
```

Alternatively, you can create static images of your model with the more flexible `swmmio.draw_model` function. With that, you could highlight specific conduits by giving them a distinct color, which looks something like this:

```python 
mymodel = swmmio.Model('path/to/model.inp')
links = mymodel.links.dataframe

# provide custom colors to links by adding a `draw_color` column to 
# links of interest using Pandas .loc assignment logic
links.loc[['SewerA', 'SewerB'], 'draw_color'] = '#e319a0' 

# then use draw_model and explicitly pass in the modified links dataframe
swmmio.draw_model(nodes=mymodel.nodes.dataframe, conduits=links)
```

Hopefully this helps! Let us know how you make out :-)

**Reply by bemcdonnell** [reply]
@CedarTH did @aerispaha's suggestion work for you? If so, i will mark it as complete.

**Reply by CedarSnowy** [reply]
Thank you all!! I have finished my work by using mapbox,and @aerispaha 's method is useful.

**Reply by bemcdonnell** [reply]
@CedarTH awesome!! If you want to put a screenshot up here to help future users, we’d love that :-)
